SecurityPkg/Library/PeiTpmMeasurementLib/PeiTpmMeasurementLib.c

```diff
@@ -44,31 +107,31 @@ TpmMeasureAndLogData (
   IN UINT64  HashDataLen
   )
 {
-  EFI_STATUS         Status;
-  EDKII_TCG_PPI      *TcgPpi;
-  TCG_PCR_EVENT_HDR  TcgEventHdr;
+  EFI_STATUS     Status;
+  EDKII_TCG_PPI  *TcgPpi;
+  EDKII_CC_PPI   *CcPpi;
+
+  Status = PeiServicesLocatePpi (
+             &gEdkiiCcPpiGuid,
+             0,
+             NULL,
+             (VOID **)&CcPpi
+             );
+  if (!EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "PeiTpmMeasureAndLogData with Cc Measurement Ppi \n"));
+    return CcMeasureAndLogData (CcPpi, PcrIndex, EventType, EventLog, LogLen, HashData, HashDataLen);

```

Comment:
Why return here?

In case a confidential VM has a vTPM I'd say we should measure to both CC PCRs and vTPM.

Reply:
This needs clarification. AIUI, the CC and TPM measurements were not intended to be combined, and on platforms that implement CC, the vTPM is backed by the untrusted VMM.

There is also only a single TPM event log for both, and it does not specify whether measurements were taken into the CC RTMRs or into the TPM PCRs.

@jyao1 any thoughts?

Reply:
In current OVMF implementation CC and TPM measurements are not combined together, which means only ONE measurement is called in one time. This PR updates PeiTpmMeasurementLib to support CC measurement and it keeps consistent with the current implementation.

I think we can submit another PR to address above discussion (whether to support CC and TPM measurement together or not) once there is a decision.

Reply:
Agreed.

CC and vTPM are both exposed by the VM's system firmware. So if a system implements both, I would expect it to be the system firmware's responsibility to keep the CC registers updated as the vTPM measurements are taken, rather than make it the job of each component running in the boot services context to be aware of this and measure into both.